hedera-node/src/main/java/com/hedera/services/store/TypedTokenStore.java

```diff
@@ -271,24 +245,30 @@ public void persistToken(Token token) {
 		mapModelChangesToMutable(token, mutableToken);
 
 		if (token.hasMintedUniqueTokens()) {
-			final var currentNfts = nfts.get();
+			final var currentNfts = uniqueTokens.get();
+			final var currentUniqueTokenAssociations = uniqueTokenAssociations.get();
+			final var currentUniqueTokenAccountOwnerships = uniqueOwnership.get();
 			for (var uniqueToken : token.mintedUniqueTokens()) {
 				final var mintKey = new MerkleUniqueTokenId(
 						new EntityId(uniqueToken.getTokenId()), uniqueToken.getSerialNumber());
 				final var merkleUniqueToken = new MerkleUniqueToken(
 						new EntityId(uniqueToken.getOwner()), uniqueToken.getMetadata(), uniqueToken.getCreationTime());
-				uniqueTokens.get().put(mintKey, merkleUniqueToken);
 				currentNfts.put(mintKey, merkleUniqueToken);
+				currentUniqueTokenAccountOwnerships.associate(new EntityId(uniqueToken.getOwner()), mintKey);
+				currentUniqueTokenAssociations.associate(new EntityId(uniqueToken.getTokenId()), mintKey);
 				backingNfts.addToExistingNfts(mintKey.asNftId());
 			}
 		}
 		if (token.hasBurnedUniqueTokens()) {
-			final var currentNfts = nfts.get();
+			final var currentNfts = uniqueTokens.get();
+			final var currentUniqueTokenAssociations = uniqueTokenAssociations.get();
+			final var currentUniqueTokenAccountOwnerships = uniqueOwnership.get();
 			for (var uniqueToken : token.burnedUniqueTokens()) {
 				final var burnKey = new MerkleUniqueTokenId(
 						new EntityId(uniqueToken.getTokenId()), uniqueToken.getSerialNumber());
-				uniqueTokens.get().remove(burnKey);
 				currentNfts.remove(burnKey);
+				currentUniqueTokenAccountOwnerships.disassociate(new EntityId(uniqueToken.getOwner()), burnKey);
```

Comment:
I think that we are not populating `owner` for the `uniqueToken` on burn. It would be best to directly use the `token.treasury` for the `new EntityId()` constructor. If we do that for `burn` it would be best to do it in `mint` for consistency